src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs

```diff
@@ -114,5 +128,108 @@ internal static bool TryGetPathOfFileAbove(string file, DirectoryInfo startingDi
 
             return false;
         }
+
+        /// <summary>
+        /// Parses the <c>msbuild-sdks</c> section of the specified JSON string.
+        /// </summary>
+        /// <param name="json">The JSON to parse as a string.</param>
+        /// <returns>A <see cref="Dictionary{TKey, TValue}" /> containing MSBuild project SDK versions if any were found, otherwise <c>null</c>.</returns>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Dictionary<string, string> ParseMSBuildSdkVersionsFromJson(string json)
+        {
+            Dictionary<string, string> versionsByName = null;
+
+            using (var reader = new JsonTextReader(new StringReader(json)))
+            {
+                if (!reader.Read() || reader.TokenType != JsonToken.StartObject)
+                {
+                    return null;
+                }
+
+                while (reader.Read())
+                {
+                    if (reader.TokenType == JsonToken.PropertyName && reader.Value is string objectName && string.Equals(objectName, MSBuildSdksPropertyName, StringComparison.Ordinal) && reader.Read() && reader.TokenType == JsonToken.StartObject)
+                    {
+                        while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+                        {
+                            if (reader.TokenType == JsonToken.PropertyName && reader.Value is string name && reader.Read() && reader.TokenType == JsonToken.String && reader.Value is string value)
+                            {
+                                versionsByName ??= new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+                                versionsByName[name] = value;
+
+                                continue;
+                            }
+
+                            reader.Skip();
+                        }
+
+                        return versionsByName;
+                    }
+
+                    // Skip any top-level entry that's not a property
+                    reader.Skip();
+                }
+            }
+
+            return versionsByName;
+        }
+
+        /// <summary>
+        /// Fires the <see cref="FileRead" /> event for the specified file.
+        /// </summary>
+        /// <param name="filePath">The full path to file that was read.</param>
+        private void OnFileRead(string filePath)
+        {
+            EventHandler fileReadEventHandler = FileRead;
+
+            fileReadEventHandler?.Invoke(filePath, EventArgs.Empty);
```

Comment:
This looks like non-standard usage passing an event argument for `sender`.  How about this?
```C#
public event EventHandler<string> FileRead;

...

fileReadEventHandler?.Invoke(this, filePath);
```

Reply:
Thanks, I was looking for something like this!